version: '3.8'

services:

  manager:
    build:
      context: ./manager
      dockerfile: Dockerfile
    expose:
      - "3000"
    ports:
      - "3000:3000"
    extra_hosts:
      - "host.docker.internal:172.17.0.1"


  client:
    build:
      context: ./client
      dockerfile: Dockerfile
    depends_on:
      - manager
    restart: on-failure
    deploy:
      mode: replicated
      replicas: 2

  monitor:
    build:
      context: ./monitor
      dockerfile: Dockerfile
    depends_on:
      - manager
      - client
    expose:
      - "3006"
    ports:
      - "3006:3006"

  mongo:
    image: mongo:4.4.12 # Last 4.x, latest image is 5+ incompatible with mongodb k8s version
    volumes:
      - ./be/mongodb-create-user.sh:/docker-entrypoint-initdb.d/mongodb-create-user.sh
    env_file: .env
    environment:
      MONGO_INITDB_ROOT_USERNAME: ${MONGO_ROOT_USERNAME}
      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_ROOT_PASSWORD}
      MONGO_INITDB_DATABASE: ${MONGO_DB_DATABASE}
      MONGO_DB_DATABASE: ${MONGO_DB_DATABASE}
      MONGO_DB_USERNAME: ${MONGO_DB_USERNAME}
      MONGO_DB_PASSWORD: ${MONGO_DB_PASSWORD}
    ports:
      - "27017:27017"
    healthcheck:
      test: echo 'db.runCommand("ping").ok' | mongo mongodb:27017/${MONGO_DB_DATABASE} --quiet
      interval: 30s
      timeout: 5s
      start_period: 10s
      retries: 3

